source/Applications/Advanced/TransformPointCloudViaArucoMarker/TransformPointCloudViaArucoMarker.cpp

```diff
@@ -0,0 +1,313 @@
+        }
+
+        cv::Mat newRotation = rotation.inv();
+        cv::Mat newTranslation = -newRotation * translation;
+
+        auto cvInverted = cv::Mat(4, 4, CV_32F, 0.0);
+        for(int i = 0; i < 3; ++i)
+        {
+            for(int j = 0; j < 3; ++j)
+            {
+                cvInverted.at<float>(i, j) = newRotation.at<float>(i, j);
+            }
+        }
+        for(int i = 0; i < 3; ++i)
+        {
+            cvInverted.at<float>(i, 3) = newTranslation.at<float>(i, 0);
+        }
+        cvInverted.at<float>(3, 3) = 1.0;
+
+        auto *ptr = &cvInverted.at<float>(0, 0);
+        auto invertedTransform = Zivid::Matrix4x4(ptr, ptr + 16);
+
+        return invertedTransform;
+    }
+
+	cv::Mat pointCloudToColorBGR(const Zivid::PointCloud &pointCloud)
+    {
+        auto rgb = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        pointCloud.copyData(
+            reinterpret_cast<Zivid::ColorRGBA *>(rgb.data)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgb, bgr, cv::COLOR_RGBA2BGR);
+
+        return bgr;
+    }
+
+	cv::Mat colorBGRToGray(const cv::Mat &bgr)
+	{
+        cv::Mat gray;
+        cv::cvtColor(bgr, gray, cv::COLOR_BGR2GRAY);
+        return gray;
+	}    
+
+	void displayBGR(const cv::Mat &bgr, const std::string &bgrName)
+    {
+        cv::namedWindow(bgrName, cv::WINDOW_AUTOSIZE);
+        cv::imshow(bgrName, bgr);
+        cv::waitKey(0);
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        const auto arucoMarkerFile = std::string(ZIVID_SAMPLE_DATA_DIR) + "/ArucoMarkerInCameraOrigin.zdf";
+        std::cout << "Reading ZDF frame from file: " << arucoMarkerFile << std::endl;
+        const auto frame = Zivid::Frame(arucoMarkerFile);
+        auto pointCloud = frame.pointCloud();
+		
+		std::cout << "Converting to OpenCV image format" << std::endl;
+        const auto bgrImage = pointCloudToColorBGR(pointCloud);
+        const auto grayImage = colorBGRToGray(bgrImage);
+        
+        std::cout << "Configuring ArUco marker" << std::endl;
+        const auto markerDictionary = cv::aruco::getPredefinedDictionary(cv::aruco::DICT_4X4_100);
+        std::vector<int> markerIds;
+        std::vector<std::vector<cv::Point2f>> markerCorners;
+        cv::Ptr<cv::aruco::DetectorParameters> detectorParameters = cv::aruco::DetectorParameters::create();
+        detectorParameters->cornerRefinementMethod = cv::aruco::CORNER_REFINE_SUBPIX;
+
+        std::cout << "Detecting ArUco Marker" << std::endl;
+        cv::aruco::detectMarkers(grayImage, markerDictionary, markerCorners, markerIds, detectorParameters);
+
+		if(markerIds.empty())
+        {
+            std::cout << "No ArUco markers detected";
+            exit(EXIT_SUCCESS);
+        }
```

Comment:
I think you should also error out if you detect more than one marker, since then it's ambiguous what to do next. I see you currently just use the marker at index 0, but that's kind of arbitrary.

Reply:
I agree with moving this check after displaying the marker. That is done.

What if a customer has a scene with two markers?
In this state of the code sample, it will work (on a random marker, yes).
If I implement that only one marker in the scene is necessary for the sample to work, then a customer needs to cover the other markers or move them out of the scene, which I don't like.

What do you think?

Reply:
You have good points. You can leave it as is, and allow multiple markers.

Have you checked that "cv::aruco::drawDetectedMarkers" doesn't crash or something if there are no markers?

Reply:
Yes, it just returns empty as it should.

Reply:
I haven't done that. Good point.

Reply:
Tested without marker and it works fine. 
One thing. Add "endl":
```
std::cout << "No ArUco markers detected" << std::endl;
```

Reply:
Also you pushed the commented-away display line :)

Reply:
fixed 👍